fuel-tx/src/transaction/types/chargeable_transaction.rs

```diff
@@ -0,0 +1,388 @@
+use crate::{
+    field::ChargeableBody,
+    policies::Policies,
+    transaction::{
+        field::{
+            Inputs,
+            Outputs,
+            Policies as PoliciesField,
+            Witnesses,
+        },
+        id::PrepareSign,
+        metadata::CommonMetadata,
+        validity::{
+            check_common_part,
+            FormatValidityChecks,
+        },
+        Chargeable,
+    },
+    ConsensusParameters,
+    Input,
+    Output,
+    UniqueIdentifier,
+    ValidityError,
+    Witness,
+};
+use derivative::Derivative;
+use fuel_types::{
+    bytes,
+    canonical::Serialize,
+    BlockHeight,
+    Bytes32,
+    ChainId,
+};
+use hashbrown::HashMap;
+
+#[cfg(feature = "alloc")]
+use alloc::vec::Vec;
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct ChargeableMetadata<Body> {
+    pub common: CommonMetadata,
+    pub body: Body,
+}
+
+#[derive(Clone, Derivative)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(fuel_types::canonical::Deserialize, fuel_types::canonical::Serialize)]
+#[derivative(Eq, PartialEq, Hash, Debug)]
+pub struct ChargeableTransaction<Body, MetadataBody> {
+    pub(crate) body: Body,
```

Comment:
The change is breaking because now old fields live under the `body` subfield. We could avoid it by flattening it, but it is [not supported](https://github.com/bincode-org/bincode/issues/245) properly by serde for the bincode.